addon/components/bs-contextual-help.js

```diff
@@ -191,10 +207,7 @@ export default class ContextualHelp extends Component {
    * @readonly
    * @private
    */
-  @computed
-  get overlayId() {
-    return `overlay-${guidFor(this)}`;
-  }
+  overlayId = `overlay-${guidFor(this)}`;
```

Comment:
Why aren't you using a getter here?

```suggestion
  get overlayId() {
    return `overlay-${guidFor(this)}`;
  }
```

The current syntax allows `overlayId` to be set. But it should be read-only, shouldn't it? Are you afraid of performance issues if not caching it? I would have assumed that `guidFor()` util uses a cache and therefore doesn't have a relevant performance impact.

Reply:
It's basically a constant value, as it never changes after component instantiation, so I don't see a value in having a getter (that *could* react to changes over time) instead of a more simple initializer. 🤷‍♂️

Reply:
Seems we are missing a const property per class instances. So both styles have trade-offs. Maybe it would be good to document that these property should be considered read-only? I got confused by that syntax cause I assumed that you aren't using a getter cause it could be set somewhere.

Reply:
> Seems we are missing a const property per class instances

What do you mean by this?

> Maybe it would be good to document that these property should be considered read-only?

It is! 😀  Line 207. And it's private anyway, so shouldn't be used outside of the class.

Reply:
> > Seems we are missing a const property per class instances
> 
> What do you mean by this?

I imagine a language feature here, which does not exist as far as I know. Maybe something like:

```js
class Foo {
  const id = guidFor(this);
}
```

A property defined as `const` could not be changed. Trying to set it would throw an exception. So it would be similar to a native getter without a setter in that regard.

But other than a native getter it should be guaranteed to not change over time. So it would be similar to a const variable in that regard.

Other than a static class property the value would be assigned per instance. So it would not be the same for all instances of that class. Additionally it could depend on that instance. As shown in the example.

But I'm getting off topic here. 🙈 As it's already documented as read-only (thanks for pointing it out) I will close this thread as resolved.